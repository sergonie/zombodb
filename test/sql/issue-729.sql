CREATE TABLE tst_pagination
(
    id    SERIAL
        CONSTRAINT tst_pagination_pk PRIMARY KEY,
    title VARCHAR
);

CREATE INDEX idx_tst_pagination
    ON tst_pagination
    USING zombodb ((tst_pagination.*));

INSERT INTO tst_pagination (title)
VALUES ('Fruits: orange'),
       ('Fruits: mango'),
       ('Fruits: apple'),
       ('Fruits: pear'),
       ('Fruits: kiwi');

--without offset - its ok
SELECT zdb.score(ctid) > 0 as score,
       id,
       title
FROM tst_pagination
WHERE tst_pagination ==> dsl.limit(2, dsl.offset(0, dsl.sort('_score', 'desc', 'fruits'::text)))
ORDER BY id;

-- with offset with score = 0 :(
SELECT zdb.score(ctid) > 0 as score,
       id,
       title
FROM tst_pagination
WHERE tst_pagination ==> dsl.limit(2, dsl.offset(2, dsl.sort('_score', 'desc', 'fruits'::text)))
ORDER BY id;

DROP TABLE tst_pagination CASCADE;