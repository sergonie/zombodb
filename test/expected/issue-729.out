CREATE TABLE tst_pagination
(
    id    SERIAL
        CONSTRAINT tst_pagination_pk PRIMARY KEY,
    title VARCHAR
);
CREATE INDEX idx_tst_pagination
    ON tst_pagination
    USING zombodb ((tst_pagination.*));
INSERT INTO tst_pagination (title)
VALUES ('Fruits: orange'),
       ('Fruits: mango'),
       ('Fruits: apple'),
       ('Fruits: pear'),
       ('Fruits: kiwi');
--without offset - its ok
SELECT zdb.score(ctid) > 0 as score,
       id,
       title
FROM tst_pagination
WHERE tst_pagination ==> dsl.limit(2, dsl.offset(0, dsl.sort('_score', 'desc', 'fruits'::text)))
ORDER BY id;
 score | id |     title     
-------+----+---------------
 t     |  2 | Fruits: mango
 t     |  4 | Fruits: pear
(2 rows)

-- with offset with score = 0 :(
SELECT zdb.score(ctid) > 0 as score,
       id,
       title
FROM tst_pagination
WHERE tst_pagination ==> dsl.limit(2, dsl.offset(2, dsl.sort('_score', 'desc', 'fruits'::text)))
ORDER BY id;
 score | id |     title      
-------+----+----------------
 t     |  1 | Fruits: orange
 t     |  3 | Fruits: apple
(2 rows)

DROP TABLE tst_pagination CASCADE;
